### PR TITLE
[BB-5559] Create the plugin to store LTI configurations

### PR DESCRIPTION
**Pull request by tecoholic**
### Description

This PR establishes the `openedx_ltistore` plugin that can store LTI configurations centrally in the database and make it available for all the courses.

### Testing instructions

Since this is the first code commit to the repository, it can be tested by following the instructions in the [README](https://github.com/open-craft/openedx-ltistore/tree/tecoholic/BB-5559-centrally-store-lti-keys#readme) file.

To test the integration with the LTI Consumer XBlock, use the [open-craft:tecoholic/BB-5559-central-stroage-of-lti-keys](https://github.com/open-craft/xblock-lti-consumer/tree/tecoholic/BB-5559-central-stroage-of-lti-keys) branch and install it locally.

### Additional information

1. [OpenCraft ticket](https://tasks.opencraft.com/browse/BB-5559)
1. [LTI Consumer XBlock PR](https://github.com/openedx/xblock-lti-consumer/pull/239)

### CODE REVIEW

File: README.md
Comment (giovannicimolin) [inline README.md:23]: This should not be needed if this has the proper hooks in place. Pluggable apps do this automatically.
Checkout the code in the proctoring app:

- https://github.com/openedx/edx-proctoring/blob/6a21189c6e8c2ac421fe0e9755daf9a7a7a56916/setup.py#L93
- https://github.com/openedx/edx-proctoring/blob/6a21189c6e8c2ac421fe0e9755daf9a7a7a56916/edx_proctoring/apps.py#L60

We also have a few client pluggable apps in GitLab, take a look at them as well.

File: openedx_ltistore/serializers.py
Comment (giovannicimolin) [inline openedx_ltistore/serializers.py:1]: Remove this dependency to `rest_framework`. Do you think you can use `from django.core import serializers
` do to the job here?

**Review by giovannicimolin — CHANGES_REQUESTED** [review]

File: openedx_ltistore/serializers.py
Comment (tecoholic) [reply openedx_ltistore/serializers.py:1]: I replaced this with `django.forms.models.model_to_dict`

File: README.md
Comment (tecoholic) [reply README.md]: @giovannicimolin Thank you for the pointers. Implemented the plugin mechanism based on the Proctering app.

File: lti_store/models.py
Comment (giovannicimolin) [inline lti_store/models.py:72]: We only support LTI 1.1 configurations, for now, you can remove these fields until we implement the new version.
```suggestion
```

**Review by giovannicimolin — CHANGES_REQUESTED** [review]

File: README.md
Comment (giovannicimolin) [inline README.md:26]: @tecoholic This also needs to be installed in the LMS env, otherwise it'll fail to launch when published

File: lti_store/apps.py
Comment (tecoholic) [reply lti_store/apps.py:15]: Yes. I added this following the Proctoring app. We can remove this.

File: lti_store/models.py
Comment (tecoholic) [reply lti_store/models.py]: Done. The plugin now uses a slug field and uses the slug for the filter key as well.

**Comment by tecoholic** [comment]
@giovannicimolin I have refactored the plugin based on your review feedback and updated the README as well. Give it another pass when you get time. Thank you.

File: lti_store/models.py
Comment (giovannicimolin) [inline lti_store/models.py:7]: @tecoholic Remove the LTI 1.3 version choice from here for now.

**Review by giovannicimolin — APPROVED** [review]
:+1:

- [x] I tested this: tested alongside https://github.com/openedx/xblock-lti-consumer/pull/239
- [x] I read through the code
- [ ] I checked for accessibility issues
- [x] Includes documentation

@tecoholic Looks good. Left a minor comment on the PR. 
Good to merge once that is addressed.

**Comment by giovannicimolin** [comment]
@tecoholic Remember to squash your commits here before merging.

File: lti_store/models.py
Comment (tecoholic) [reply lti_store/models.py]: Done :)

**Comment by tecoholic** [comment]
@giovannicimolin I have updated the PR by removing the version 1.3 choice. I am merging to master.

**Pull request merged**
